test/e2e/parallel/spacerequest_test.go

```diff
@@ -80,28 +93,50 @@ func TestCreateSpaceRequest(t *testing.T) {
 				// then
 				// spaceRequest should reset back the tierName
 				_, err = awaitilities.Host().WaitForSpace(t, subSpace.GetName(),
-					UntilSpaceHasTier("appstudio"), // tierName is back as the one on spaceRequest
+					UntilSpaceHasTier("appstudio-env"), // tierName is back as the one on spaceRequest
 					UntilSpaceHasTargetClusterRoles(targetClusterRoles),
 					UntilSpaceHasAnyProvisionedNamespaces(),
 				)
 				require.NoError(t, err)
 
-				t.Run("delete space request", func(t *testing.T) {
-					// now, delete the SpaceRequest and expect that the Space will be deleted as well,
-					// along with its associated namespace
-
+				t.Run("namespace access secret is recreated if deleted ", func(t *testing.T) {
+					// now, delete the namespace access secret
 					// when
-					err := memberAwait.Client.Delete(context.TODO(), spaceRequest)
+					err := memberAwait.Client.Delete(context.TODO(), namespaceAccessSecret)
 
 					// then
-					// subSpace should be deleted as well
 					require.NoError(t, err)
-					err = memberAwait.WaitUntilNamespaceDeleted(t, subSpace.Name, "appstudio")
+					err = memberAwait.WaitUntilSecretDeleted(t, namespaceAccessSecret)
 					require.NoError(t, err)
-					err = memberAwait.WaitUntilNSTemplateSetDeleted(t, subSpace.Name)
-					require.NoError(t, err)
-					err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(t, subSpace.Name)
+					// a new secret should be created
+					spaceRequest, err = memberAwait.WaitForSpaceRequest(t, types.NamespacedName{Namespace: spaceRequest.GetNamespace(), Name: spaceRequest.GetName()},
+						UntilSpaceRequestHasConditions(Provisioned()),
+						UntilSpaceRequestHasStatusTargetClusterURL(memberCluster.Spec.APIEndpoint),
+						UntilSpaceRequestHasNamespaceAccess(),
+					)
 					require.NoError(t, err)
+					// verify secret containing kubeconfig was provisioned
+					kubeClient, _ = newKubeClientFromSecret(t, memberAwait, spaceRequest.Namespace, spaceRequest.Status.NamespaceAccess[0].SecretRef)
+					// check that kubeconfig is valid
+					validateKubeClient(t, kubeClient, spaceRequest)
+
+					t.Run("delete space request", func(t *testing.T) {
+						// now, delete the SpaceRequest and expect that the Space will be deleted as well,
+						// along with its associated namespace
+
+						// when
+						err := memberAwait.Client.Delete(context.TODO(), spaceRequest)
+
+						// then
+						// subSpace should be deleted as well
+						require.NoError(t, err)
+						err = memberAwait.WaitUntilNamespaceDeleted(t, subSpace.Name, "appstudio-env")
+						require.NoError(t, err)
+						err = memberAwait.WaitUntilNSTemplateSetDeleted(t, subSpace.Name)
+						require.NoError(t, err)
+						err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(t, subSpace.Name)
+						require.NoError(t, err)
+					})
```

Comment:
could you also verify that the secret is deleted?

Reply:
Yes we could,  but I tought that once we verify that the namespace is deleted, all resources in there will be deleted 🤷‍♂️ . Am I missing anything ?

Reply:
I was referring to the Secret that was created by the controller next to the SpaceRequest CR - that's a different namespace.

Reply:
Ok got it! I've added a check that verifies if all secrets are deleted as well. Plz see: https://github.com/codeready-toolchain/toolchain-e2e/pull/706/commits/98dc38544abbbb03436fd99b49ca78a10aba2254